src/ksync/ksync_sock.cc

```diff
@@ -554,16 +600,51 @@ KSyncSockNetlink::KSyncSockNetlink(boost::asio::io_service &ios, int protocol)
     boost::system::error_code ec1;
     sock_.get_option(rcv_buf_size, ec);
     LOG(INFO, "Current receive sock buffer size is " << rcv_buf_size.value());
-#endif
 }
+#else
+KSyncSockNetlink::KSyncSockNetlink(boost::asio::io_service &ios)
+    : pipe_(ios)
+{
+    DWORD access_flags = GENERIC_READ | GENERIC_WRITE;
+    DWORD attrs = OPEN_EXISTING;
+
+    nl_client_->cl_win_pipe = CreateFile(KSYNC_PATH, access_flags, 0, NULL, attrs, 0, NULL);
+    if (nl_client_->cl_win_pipe == INVALID_HANDLE_VALUE) {
+        LOG(ERROR, "Error while opening KSync pipe: " << GetFormattedWindowsErrorMsg());
+        assert(0);
```

Comment:
I think these `asserts` are misused here. Opening the file can fail for other reasons than a programmer error (e.g. extension was not enabled). When project is compiled using release settings, assert macros are compiled to empty instructions, thus an agent will not terminate. I think `exit` or `abort` should be used here.